application/core/FrameworkCore.php

```diff
@@ -9,278 +10,283 @@
-						$this->controller->{$_instance}->view = null;
-					}
-				}
-			}
+    public $segment, $host, $partial, $controller, $title, $description,
+    $serverPath, $root, $flash, $filePath, $uriPath, $http;
+    public $canonical   = '';
+    public $pageData    = [];
+    public $ignoredExts = [];
+    
+    protected $allowedSegments, $pageController;
+    
+    private $errorReporting, $allowedFileExts;
+    
+    /**
+     * Core constructor
+     * 
+     * @param   field_type
+     * @author  sbebbington
+     * @date    26 Sep 2017 14:42:15
+     * @version 0.1.4-RC3
+     * @return  void
+     * @throws  FrameworkException
+     */
+    public function __construct(){
+        HtmlBuilder::__construct();
+        if($this->setSiteConfiguration() == false){
+            throw new FrameworkException("Please set up a pages.json file in the config folder", "0x00");
+        }
+        $this->pageData     = $this->getPageData();
+        
+        $this->setErrorReporting();
+        $this->setUri();        
+        $this->setGetGlobal();
+        $this->checkExtension();
+        
+        $this->serverPath   = serverPath();        
+        $this->root         = str_replace("\\", "/", $_SERVER['DOCUMENT_ROOT']);
+        $this->controller   = new stdClass();
+        $this->flash        = new stdClass();
+        $this->partial      = array(
+            'header'    => (file_exists(serverPath("/view/partial/header.phtml"))) ? serverPath("/view/partial/header.phtml") : '',
+            'footer'    => (file_exists(serverPath("/view/partial/footer.phtml"))) ? serverPath("/view/partial/footer.phtml") : '',
+        );
+    }
+    
+    /**
+     * Sets up the site configuration according
+     * to the JSON objects in 
+     * ../application/config/pages.json
+     * Note that allowedSegments and pageControllers
+     * are required settings
+     * 
+     * @param   na
+     * @author  sbebbington
+     * @date    28 Jul 2017 14:29:45
+     * @version 0.1.4-RC3
+     * @return  boolean
+     * @throws  FrameworkException
+     */
+    protected function setSiteConfiguration(){
+        if(!file_exists(serverPath('/config/pages.json'))){
+            return false;
+        }
+        $siteConfiguration      = json_decode(file_get_contents(serverPath('/config/pages.json')), true);
+        $this->allowedSegments  = $siteConfiguration['allowedSegments'];
+        $this->pageController   = $siteConfiguration['pageController'];
+        $this->errorReporting   = $siteConfiguration['errorReporting'] ?? [];
+        $this->allowedFileExts  = $siteConfiguration['allowedFileExts'] ?? [];
+        $this->ignoredExts      = $siteConfiguration['ignotedFileExts'] ?? ['js', 'css'];
+        
+        if(!empty($this->allowedSegments) && !empty($this->pageController)){
+            return true;
+        }
+        throw new FrameworkException("No pages or page controllers set in the pages.json file", "0x01");
```

Comment:
I would do this the other way round ie if condition not met throw exception else return true

Reply:
Maybe my mind works backwards.